Wrappers/Python/cil/framework/framework.py

```diff
@@ -2530,11 +2532,9 @@ def __init__(self,
         if array is None:                                   
             array = numpy.empty(geometry.shape, dtype=dtype)
         elif issubclass(type(array) , DataContainer):
-            array = array.as_array()
-        elif issubclass(type(array) , numpy.ndarray):
-            pass
-        else:
-            raise TypeError('array must be a CIL type DataContainer or numpy.ndarray got {}'.format(type(array)))
+            array = array.as_array()        
+        elif not all(getattr(array, attrib) for attrib in ["shape", "ndim","dtype"]):
```

Comment:
I do not understand this requirement. It is a requirement for dask, but when do we use?